src/Microsoft.VisualStudio.SlowCheetah.VS/NugetHandler/SlowCheetahNuGetManager.cs

```diff
@@ -162,7 +162,7 @@ public void CheckSlowCheetahInstallation(IVsHierarchy hierarchy)
                 };
             }
 
-            plan.Execute(currentProject);
+            this.package.JoinableTaskFactory.Run(() => plan.Execute(currentProject));
```

Comment:
Yes, this is exactly how you perform async work from the mainthread. Performing that here is fine. Alternatively you can even make this `Async` and have the caller perform the JTF.Run. Totally up to you. I personally would pull the `async` up as far as you possibly can, which would be until you hit a non-async contract you have to satisfy. Which in this case is the `OnInvoke` of your commands.